api/src/main/scala/Geoprocessing.scala

```diff
@@ -48,6 +48,69 @@ trait Geoprocessing extends Utils {
   }
 
   /**
+    * For an InputData object, return a histogram of raster lines join results.
+    *
+    * @param   input  The InputData
+    * @return         A histogram of results
+    */
+  def getRasterLinesJoin(input: InputData): ResultInt = {
+    val aoi = createAOIFromInput(input)
+    val rasterLayers = cropRastersToAOI(input.rasters, input.zoom, aoi)
+    val lines = input.vector match {
+      case Some(vector) =>
+        input.vectorCRS match {
+          case Some(crs) =>
+            createMultiLineFromInput(vector, crs, input.rasterCRS)
+          case None =>
+            throw new Exception("Request data missing required 'vectorCRS'.")
+        }
+      case None =>
+        throw new Exception("Request data missing required 'vector'.")
+    }
+    ResultInt(rasterLinesJoin(rasterLayers, lines))
+  }
+
+  /**
+    * Given a collction of rasterLayers & a collection of lines, return the
+    * values intersected by the rasterized lines.
+    *
+    * @param   rasterLayers  A sequence of TileLayerCollections
+    * @param   lines         A sequence of MultiLines
+    * @return                A map of pixel counts
+    */
+  private def rasterLinesJoin(
+    rasterLayers: Seq[TileLayerCollection[SpatialKey]],
+    lines: Seq[MultiLine]
+  ): Map[String, Int] = {
+    val metadata = rasterLayers.head.metadata
+    var pixelGroups: TrieMap[(List[Int], List[Int], SpatialKey), Int] = TrieMap.empty
+
+    joinCollectionLayers(rasterLayers).par
+      .foreach({ case (key, tiles) =>
+        val extent = metadata.mapTransform(key)
+        val re = RasterExtent(extent, metadata.layout.tileCols,
+            metadata.layout.tileRows)
+
+        lines.par.foreach({ multiLine =>
+          Rasterizer.foreachCellByMultiLineString(multiLine, re) { case (col, row) =>
+            val pixelGroup: (List[Int], List[Int], SpatialKey) =
+              (tiles.map(_.get(col, row)).toList, List(col, row), key)
+            pixelGroups.getOrElseUpdate(pixelGroup, 1)
+          }
+        })
+      })
+
+    pixelGroups
+      .toSeq
```

Comment:
This bit of code can be simplified to this:

```diff
diff --git a/api/src/main/scala/Geoprocessing.scala b/api/src/main/scala/Geoprocessing.scala
index 3fb8e3e..7128dce 100644
--- a/api/src/main/scala/Geoprocessing.scala
+++ b/api/src/main/scala/Geoprocessing.scala
@@ -101,13 +101,8 @@ trait Geoprocessing extends Utils {
       })
 
     pixelGroups
-      .toSeq
-      .map { case ((key, _, _), value) => (key, value) }
-      .groupBy(_._1)
-      .map { case (k, v) =>
-        k.toString -> v.map { case (_, n) => n }.sum
-      }
-      .toMap
+      .groupBy(_._1._1.toString)
+      .mapValues(_.size)
   }
 
   /**
```

Reply:
Nice!